src/main/java/org/broadinstitute/hellbender/tools/spark/ParallelCopyGCSDirectoryIntoHDFSSpark.java

```diff
@@ -0,0 +1,171 @@
+package org.broadinstitute.hellbender.tools.spark;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.AuthHolder;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import scala.Tuple2;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This tool uses Spark to do a parallel copy of either a file or a directory from GCS into HDFS.
+ * Files are divided into chunks of size equal to the HDFS block size (with the exception of the final
+ * chunk) and each Spark task is responsible for copying one chunk. To copy all of the files in a GCS directory,
+ * provide the GCS directory path, including the trailing slash. Directory copies are non-recursive so
+ * subdirectories will be skipped. Within directories each file is divided into chunks independently (so this will be
+ * inefficient if you have lots of files smaller than the block size). After all chunks are copied, the HDFS
+ * concat method is used to stitch together chunks into single files without re-copying them.
+ */
+@CommandLineProgramProperties(summary="Parallel copy a file or directory (non-recursive) from GCS into HDFS",
+        oneLineSummary="Parallel copy a file or directory from GCS into HDFS.",
+        programGroup = SparkProgramGroup.class)
+public class ParallelCopyGCSDirectoryIntoHDFSSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc = "input GCS file path (add trailing slash when specifying a directory)", fullName = "inputGCSPath")
+    private String inputGCSPath = null;
+
+
+    @Argument(doc = "output directory on HDFS", shortName = "outputHDFSDirectory",
+            fullName = "outputHDFSDirectory")
+    private String outputHDFSDirectory;
+
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final AuthHolder auth = getAuthHolder();
+        final String inputGCSPathFinal = inputGCSPath;
+        final String outputDirectoryFinal = outputHDFSDirectory;
+
+        org.apache.hadoop.fs.Path file = new org.apache.hadoop.fs.Path(outputHDFSDirectory);
+
+        try(FileSystem fs = file.getFileSystem(new Configuration())){
+
+            fs.mkdirs(file);
```

Comment:
I added a UserException if the directory already exists. Otherwise this tool would overwrite any files within the directory with the same name as an incoming file.